### ISSUE TITLE
userid and profiles merging

### ISSUE THREAD

**Issue by voarsh2** [matomo-org/matomo#15485]
My issue:

User browses the site and has 80+ returning visits. They have a userid set, and I can see all of their histories with their userid

They open an incognito browser session, 2 returning visits.

Matomo shows this as two sessions.

The incognito browser logins and we can see their userid updated in Matomo.

Problem:
the non-incognito and incognito do not show up together when I browse the profile. In other words, they're not actually linked.

Is this by design? I want to see all their sessions by userid......

**Comment by tsteur** [comment]
This is by design since they are technically different visitors. We don't have a user profile feature yet.

You could basically, as a workaround, create a segment by userID and view all visits by that user.

**Comment by voarsh2** [comment]
> You could basically, as a workaround, create a segment by userID and view all visits by that user when looking at the visitor log.

Thanks for the follow up. That workaround is quite tedious indeed! I'd need to specifically choose a userid for each user I want to look into.

I think it's a very important feature, to be able to drill down by users. Isn't that what the whole userid tracking/across devices system is for? Matching sessions with users.

The next step would be getting all of their sessions into their "profile" so you can have a clean profile history for that user.

**Comment by paulbrough** [comment]
I agree with @voarsh2. I just discovered this (somewhat) new behavior today after noticing the "first visit" date didn't look right. Visits days or weeks apart from the same device keep the same visitor ID, so I'm not sure why visits from the same user ID on different devices should be any different.

**Comment by xtrilla** [comment]
I agree with @voarsh2 and @paulbrough this seems like a quite important feature. In the actual state the User ID feature is not really useful.
